ProductWarehouse.API/Controllers/ProductsController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using ProductWarehouse.API.QueryParameters;
+using ProductWarehouse.Application.Queries;
+using ProductWarehouse.Application.Responses;
+using ProductWarehouse.Domain.Entities;
+
+namespace ProductWarehouse.API.Controllers
+{
+    /// <summary>
+    /// Controller for managing product-related operations.
+    /// </summary>
+    [ApiController]
+    [Route("api/products")]
```

Comment:
[Route("api/[controller]")] this route should be in your base controller to be generic